sensing/src/main/java/com/google/android/sensing/SensorManager.kt

```diff
@@ -0,0 +1,155 @@
+ * the behavior of supported sensors.
+ */
+interface SensorManager {
+  /**
+   * Initializes a specified sensor in preparation for capturing data.
+   *
+   * @param sensorType The type of sensor to initialize (e.g., SensorType.CAMERA,
+   * SensorType.MICROPHONE)
+   * @param context Android Context for accessing system resources.
+   * @param lifecycleOwner A LifecycleOwner (typically Activity or Fragment) to tie the sensor's
+   * lifecycle.
+   * @param initConfig Sensor-specific initialization configuration.
+   * @throws IllegalStateException when
+   * 1. [sensorType] is not compatible with [initConfig]
+   * 2. [init] is called before [reset]-ting the previous capture
+   */
+  suspend fun init(
+    sensorType: SensorType,
+    context: Context,
+    lifecycleOwner: LifecycleOwner,
+    initConfig: InitConfig
+  )
+
+  /**
+   * Starts the capture process for the specified sensor. Call after [init].
+   *
+   * @param sensorType The type of sensor to start.
+   * @param captureRequest Details about the capture requirements (e.g., resolution, frame rate).
+   * @throws IllegalStateException when
+   * 1. [start] is called before [init],
+   * 2. [start] is called before [reset]-ting the previous capture,
+   * 3. [sensorType] is not compatible with the given [captureRequest]
+   */
+  suspend fun start(sensorType: SensorType, captureRequest: CaptureRequest)
+
+  /**
+   * Stops the capture process for the specified sensor.
+   *
+   * @param sensorType The type of sensor to stop.
+   */
+  suspend fun stop(sensorType: SensorType)
+
+  /**
+   * **[Optional]** Temporarily pauses the capture process for the specified sensor.
+   *
+   * @param sensorType The type of sensor to pause.
+   */
+  suspend fun pause(sensorType: SensorType)
+
+  /**
+   * **[Optional]** Resumes the capture process for a previously paused sensor.
+   *
+   * @param sensorType The type of sensor to resume.
+   */
+  suspend fun resume(sensorType: SensorType)
+
+  /**
+   * Resets SensorManager for the specified sensor, killing the sensor if its capturing and
+   * releasing any acquired resources. Call it after [stop]. If called before [stop] capturing is
+   * [kill]ed.
+   *
+   * @param sensorType The type of sensor to reset.
+   */
+  fun reset(sensorType: SensorType)
+
+  /** Interface for receiving notifications about sensor capture events and results. */
+  interface AppDataCaptureListener {
+    fun onStart(captureInfo: CaptureInfo)
+    fun onComplete(captureInfo: CaptureInfo)
+    fun onError(exception: Exception, captureInfo: CaptureInfo? = null)
+  }
+
+  /**
+   * Registers a listener to receive capture-related events for a specific sensor.
+   *
+   * @param sensorType The type of sensor for which to receive events.
+   * @param listener The AppDataCaptureListener implementation to handle events.
+   */
+  fun registerListener(sensorType: SensorType, listener: AppDataCaptureListener)
```

Comment:
when should this be called? would it be illegal to call this after sensor has started or initliazed?

Reply:
Yes you are right. Added the validations.